contracts/hardhat.config.js

```diff
@@ -46,63 +60,20 @@ task("accounts", "Prints the list of accounts", async (taskArguments, hre) => {
   return accounts(taskArguments, hre, privateKeys);
 });
 
-const isFork = process.env.FORK === "true";
-const isArbitrumFork = process.env.FORK_NETWORK_NAME === "arbitrumOne";
-const isForkTest = isFork && process.env.IS_TEST === "true";
-const isArbForkTest = isForkTest && isArbitrumFork;
-const providerUrl = `${
-  process.env.LOCAL_PROVIDER_URL || process.env.PROVIDER_URL
-}`;
-const arbitrumProviderUrl = `${process.env.ARBITRUM_PROVIDER_URL}`;
-const standaloneLocalNodeRunning = !!process.env.LOCAL_PROVIDER_URL;
+let forkBlockNumber = adjustTheForkBlockNumber();
 
-let forkBlockNumber =
-  Number(
-    isArbForkTest ? process.env.ARBITRUM_BLOCK_NUMBER : process.env.BLOCK_NUMBER
-  ) || undefined;
-if (isForkTest && standaloneLocalNodeRunning) {
-  const jsonResponse = fetch(
-    isArbForkTest ? arbitrumProviderUrl : providerUrl,
-    {
-      method: "post",
-      body: JSON.stringify({
-        jsonrpc: "2.0",
-        method: "eth_blockNumber",
-        id: 1,
-      }),
-      headers: {
-        "Content-Type": "application/json",
-      },
-    }
-  ).json();
-
-  /*
-   * We source the block number from the hardhat context rather than from
-   * node-test.sh startup script, so that block number from an already
-   * running local node can be fetched after the deployments have already
-   * been applied.
-   *
-   */
-  forkBlockNumber = parseInt(jsonResponse.result, 16);
-
-  console.log(`Connecting to local node on block: ${forkBlockNumber}`);
-
-  // Mine 40 blocks so hardhat wont complain about block fork being too recent
-  fetch(isArbForkTest ? arbitrumProviderUrl : providerUrl, {
-    method: "post",
-    body: JSON.stringify({
-      jsonrpc: "2.0",
-      method: "hardhat_mine",
-      params: ["0x28"], // 40
-      id: 1,
-    }),
-    headers: {
-      "Content-Type": "application/json",
-    },
-  }).json();
-} else if (isForkTest) {
-  console.log(`Starting a fresh node on block: ${forkBlockNumber}`);
+const paths = {};
+if (isHolesky || isHoleskyForkTest || isHoleskyFork) {
+  // holesky deployment files are in contracts/deploy/holesky
+  paths.deploy = "deploy/holesky";
+} else {
+  // holesky deployment files are in contracts/deploy/mainnet
+  paths.deploy = "deploy/mainnet";
```

Comment:
If we are going to do this, should we also split it based on the chain? Like `mainnet` for Ethereum, `arbitrum` for Arb One and `base` for Base?

One thing about splitting is that we won't be able to test all contracts at once in unit tests. However, it'll help trimming down the fixture (we already have a different fixture file for each network though)

Reply:
good point, how do you feel about it? What would you rather have?

My thinking is that we should separate and compartmentalise into separate modules things that aren't strictly connected. This should enable us (like you mention) to separate the fixture files. Also when different people work on functionalities / deploys of different chains, there should be less code overlap and thus less merge conflicts. Also tests for the chain 1 person is working on should ran faster if things are separated.